### PR TITLE
3.0.0 RC

### PR DESCRIPTION
**Pull request by cdaringe**
This PR fixes https://github.com/AmpersandJS/ampersand-select-view/issues/5.  The module and tests appeared to be designed intentionally to not fail when setting a bogus value.  I.e., if the available options are `['a', 'b', 'c']` and you call `setValue('d')`, the select would just revert to the 0th index option or unselectedText.  This behavior is dangerous.  Issue 5 and self experience would indicate the there is a bug in the application code, thus, we now trigger errors when this situation occurs.

This PR additionally:

1. Enforces that the returned value is *always a string*.  Only if `unselectedText` option is selected could `undefined` possibly be appropriate.  Would like feedback on this.  `.value` on empty `<select>`s is generally the empty string, DOM-wise.
1. Denotes a plan for 4.x release
1. adds more tests for numeric cases
1. small tidy ups

### CODE REVIEW

File: ampersand-select-view.js [hunk 1]
```diff
@@ -104,18 +104,25 @@ SelectView.prototype.onChange = function () {
...
 
     if (this.options.isCollection && this.yieldModel) {
-        value = this.findModelForId(value);
+        value = this.getModelForId(value);
     }
 
     this.setValue(value);
 };
 
-SelectView.prototype.findModelForId = function (id) {
```
Comment (cdaringe) [inline ampersand-select-view.js]: renamed to `get` prefix to fall inline with other methods.  made function `0` friendly

File: ampersand-select-view.js [hunk 2]
```diff
@@ -178,33 +185,18 @@ SelectView.prototype.clear = function() {
 };
 
 SelectView.prototype.setValue = function (value, skipValidation) {
-    if (value === this.value) return;
```
Comment (cdaringe) [inline ampersand-select-view.js]: moved 'find option' logic to own function, `getOptionByValue()`

**Comment by nickryall** [comment]
This looks good to me.  + 1

**Comment by cdaringe** [comment]
https://github.com/cdaringe/ampersand-select-view/blob/3.0.0/ampersand-select-view.js#L190 is actually a bug that needs to be addressed.  I have permitted short circuit to value `''` even if there is no `''` option available.  Will resolve and address with tests soon.

**Comment by cdaringe** [comment]
closing for rework

**Pull request closed without merging**